src/app/rundown/components/timeline-markers/timeline-markers.component.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  HostListener,
+  Input, OnChanges, SimpleChange, SimpleChanges,
+  ViewChild
+} from '@angular/core'
+import { TimestampPipe } from "../../../shared/pipes/timestamp.pipe"
+
+interface Point {
+  x: number,
+  y: number
+}
+
+const TEXT_STYLE: string = '15px arial'
+const TEXT_MIDDLE_POSITION: number = 13
+const SUBSECTION_TOP_POSITION: number = 20
+
+@Component({
+  selector: 'sofie-timeline-markers',
+  templateUrl: './timeline-markers.component.html',
+  styleUrls: ['./timeline-markers.component.scss']
+})
+export class TimelineMarkersComponent implements AfterViewInit, OnChanges {
+  @Input()
+  public time: number
+
+  @Input()
+  public pixelsPerSecond: number
+
+  @ViewChild('containerElement')
+  public containerElement: ElementRef
+
+  @ViewChild('canvasElement')
+  public canvasElement: ElementRef<HTMLCanvasElement>
+
+  private canvasContext!: CanvasRenderingContext2D
+
+  private canvasWidth: number = 0
+  private canvasHeight: number = 0
+
+  private secondsPerSection: number = 5
+  private subsectionsPerSection: number = 5
+
+  public constructor(private readonly timestampPipe: TimestampPipe) {}
+
+  @HostListener('window:resize', ['$event'])
+  public onResize(): void {
+    this.setCanvasSize()
+    this.draw()
+  }
+
+  private setCanvasSize(): void {
+    this.canvasWidth = this.containerElement.nativeElement.offsetWidth
+    this.canvasHeight = this.containerElement.nativeElement.offsetHeight
+    this.canvasElement.nativeElement.width = this.canvasWidth
+    this.canvasElement.nativeElement.height = this.canvasHeight
+  }
+
+  private draw(): void {
+    if (!this.canvasContext) {
+      return
+    }
+    this.clearCanvas()
+    this.drawSubsections()
+    this.drawSections()
+    this.drawTimestamps()
+  }
+
+  private drawSubsections(): void {
+    const subsectionWidth = this.pixelsPerSecond * this.secondsPerSection / this.subsectionsPerSection
+    const subsectionCount = this.subsectionsPerSection + Math.ceil(this.canvasWidth / subsectionWidth)
+    const timeOffsetInPixels = this.pixelsPerSecond * ((this.time / 1000) % this.secondsPerSection)
+    for (let i = 0; i < subsectionCount; i++) {
+      this.drawSubsectionIfNotCollidingWithSection(i, subsectionWidth, timeOffsetInPixels)
+    }
+  }
+
+  private drawSubsectionIfNotCollidingWithSection(subsectionIndex: number, subsectionWidth: number, timeOffsetInPixels: number): void {
+    if (subsectionIndex % this.subsectionsPerSection === 0) {
+      return
+    }
```

Comment:
You can get rid of the "if" part of this method by moving the responsibility of checking the if outside the method. Then the method only have one responsibility: Draw the section.

Reply:
I don't get rid of it, it just gets moved. I have dragged the logic for drawing the subsection out into `drawSubsectionFromIndex`.